lib/data/dynamo/table.ts

```diff
@@ -1,133 +1,140 @@
-import { DynogelsItemCallback, CreateItemOptions, Model, ModelConfiguration, Document, DocumentCollection, UpdateItemOptions, DestroyItemOptions, GetItemOptions } from 'dynogels';
+import { DynogelsItemCallback, CreateItemOptions, Model as DynModel, ModelConfiguration, Document, DocumentCollection, UpdateItemOptions, DestroyItemOptions, GetItemOptions } from 'dynogels';
 import { injectable, inject } from 'inversify';
 
 import { Dynamo } from './dynamo';
 import { MINDLESS_SERVICE_INDENTIFIERS } from '../../types';
+import {Model, ModelConstructor} from "../model";
+
 
 @injectable()
-export abstract class DynamoTable<T> {
+export abstract class DynamoTable<TModel extends Model> {
     protected abstract tableName: string;
     protected abstract definition: ModelConfiguration;
-    public model: Model;
+    public dynModel: DynModel;
 
-    constructor( @inject(MINDLESS_SERVICE_INDENTIFIERS.Dynamo) private dynamo: Dynamo) {
-    }
+    constructor(
+        @inject(MINDLESS_SERVICE_INDENTIFIERS.Dynamo) private dynamo: Dynamo,
+        protected TConstructor: ModelConstructor<TModel>
+    ) { }
 
     protected registerTable() {
-        this.model = this.dynamo.addDefinition(this.tableName, this.definition);
+        this.dynModel = this.dynamo.addDefinition(this.tableName, this.definition);
         // this.dynamo.createTables();
     }
 
-    public create(data: { [key: string]: {} }, options: CreateItemOptions = {}): Promise<T> {
+    public create(data: TModel, options: CreateItemOptions = {}): Promise<TModel> {
         let promiseCallback = (resolve, reject) => {
-            let createModelCallback: DynogelsItemCallback = (err, model) => {
+            let createModelCallback: DynogelsItemCallback = (err, document) => {
                 if (err) {
-                    console.error('Error creating a entry on ' + this.tableName + ' table. Err: ', err);
+                    console.error(`Failed to create item on ${this.tableName} table.\n Item: ${JSON.stringify(data.model)}\n Err: `, err);
                     reject(err);
                 } else {
-                    let m: T = this.transformToModel(model);
-                    resolve(m);
+                    resolve(data);
```

Comment:
Shouldn't we return something with the new document? I'm not sure if there would every be a scenario there input would be different from output for create in dynamo but I think it would make sense to make use of that constructor here. Thoughts?

Reply:
Hmm, I don't know. I feel like we should actually be returning a boolean. Can  you give me an example of when it wouldn't work? And if you are creating your code would look like this
```ts
const myModel = new myModel(data);
// do we care about the returned model?
const returnedModel = await myModelFactory.create(myModel);
// I would just ignore and continuing using myModel. Make me think create should return bool